### PR TITLE
Exercises page

### PR DESCRIPTION
**Pull request by rafaparadela**
This PR adds several changes at:

#### Routes:

- **GET /:section** Goes to the first category of the section e.g.: *http://localhost:9000/stdlib*
- **GET /:section/:category** Goes to this category of the section e.g.: *http://localhost:9000/stdlib/Extractors*

#### Model:

- **Section**: Now includes the property `color: String`
- **Exercise**: Now includes the property `explanation: Option[String]` (*Some exercises includes a clarification after the question, to improve the understanding of the concept.*)

#### Views:

- **Exercises**: The list of exercises of a particular category (*[screenshot](http://rafp.es/1QUVAH2)*)
- **Progress Bar**: Reusable progress bar to be shown at [home](http://rafp.es/1QUZn7l) and at the Exercises page.

#### ExerciseCodeExtractor:

- Ability to detect code blocks at [category description](http://rafp.es/1kmyPxb), at exercise description and at [exercise's explanation](http://rafp.es/1QUY5ZY). (And parse them to markdown).

- Ability to detect the color at [section's description](http://rafp.es/1QUYqfc).

#### Content:

- I have added two new categories (`Classes` and `Literal Booleans`) to test the navigation between them within a section. Also the parsing of its descriptions have helped to improve the `ExerciseCodeExtractor` (for instance, detecting new lines between paragraphs).

@raulraja @andyscott Could you review please? Thank you.

### CODE REVIEW

File: scala-exercises-server/app/services/parser/ExercisecodeExtractor.scala [hunk 1]
```diff
@@ -81,21 +104,52 @@ object ExerciseCodeExtractor {
       remaining match {
         case Nil ⇒ builder
         case h :: t ⇒
+
           val modBuilder = (h, builder.state) match {
             case (commentStart(startLine), DefiningClass) ⇒
               builder.addToCategoryDescription(startLine)
+            case (codeInComment(codeLine), DefiningClass) ⇒
```
Comment (raulraja) [inline scala-exercises-server/app/services/parser/ExercisecodeExtractor.scala]: All of these changes are probably in direct conflict with the work @andyscott was doing in reworking the parser using the Scala compiler API's and we need a strategy before we consider merging this. @andyscott Should we merge your stuff first and try to add this stuff later?

**Comment by raulraja** [comment]
@andyscott We are gonna get this merged today to continue progress with the frontend. Hopefully your changes to the parser can be adapted.

**Comment by andyscott** [comment]
Sorry for the slow reply.

I'm sure I can get the new exercise parser work merged in. Just don't spend
too much time fiddling with the existing parsing code.

I'll shoot to get it integrated this week.

**Comment by rafaparadela** [comment]
:+1: Sounds good. Thank you @andyscott

**Pull request merged**